funsor/torch/distributions.py

```diff
@@ -275,3 +276,5 @@ def deltadist_to_funsor(pyro_dist, output=None, dim_to_name=None):
     eager_gamma_poisson)
 eager.register(Binary, ops.SubOp, JointDirichletMultinomial, DirichletMultinomial)(  # noqa: F821
     eager_dirichlet_posterior)
+eager.register(Reduce, ops.AddOp, Multinomial[Tensor, Funsor, Funsor], frozenset)(  # noqa: F821
```

Comment:
For `pyro.collapse` purpose, it is enough to use `Multinomial[Tensor, Funsor, Tensor]` here. Then we don't have to worry about `x.value.reduce(...)` issue in `eager_plate_multinomial` above. Is there a simple solution for that and I should go ahead and replace `Funsor` by `Tensor` here in the mean time?

Reply:
>  should go ahead and replace Funsor by Tensor here in the mean time?

Sure, makes sense to me.  I don't think there's a simple solution to the plate membership indication problem, at least not one simple enough to block this PR.